actors/miner/tests/declare_recoveries.rs

```diff
@@ -0,0 +1,174 @@
+        &mut rt,
+        dl_idx,
+        p_idx,
+        BitField::try_from_bits([one_sector[0].sector_number]).unwrap(),
+        BigInt::from(0),
+    )
+    .unwrap();
+
+    let dl = h.get_deadline(&rt, dl_idx);
+    let p = dl.load_partition(&rt.store, p_idx).unwrap();
+    assert_eq!(p.faults, p.recoveries);
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn recovery_must_pay_back_fee_debt() {
+    let mut h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.set_balance(TokenAmount::from(BIG_BALANCE));
+    h.construct_and_verify(&mut rt);
+    let one_sector =
+        h.commit_and_prove_sectors(&mut rt, 1, DEFAULT_SECTOR_EXPIRATION as u64, vec![], true);
+    // advance to first proving period and submit so we'll have time to declare the fault next cycle
+    h.advance_and_submit_posts(&mut rt, &one_sector);
+
+    // Fault will take miner into fee debt
+    let mut st = h.get_state(&rt);
+    rt.set_balance(&st.pre_commit_deposits + &st.initial_pledge + &st.locked_funds);
+
+    h.declare_faults(&mut rt, &one_sector);
+
+    st = h.get_state(&rt);
+    let (dl_idx, p_idx) =
+        st.find_sector(&rt.policy, &rt.store, one_sector[0].sector_number).unwrap();
+
+    // Skip to end of proving period
+    h.advance_to_deadline(&mut rt, dl_idx);
+
+    // Can't pay during this deadline so miner goes into fee debt
+    let ongoing_pwr = power_for_sectors(h.sector_size, &one_sector);
+    let ff = pledge_penalty_for_continued_fault(
+        &h.epoch_reward_smooth,
+        &h.epoch_qa_power_smooth,
+        &ongoing_pwr.qa,
+    );
+    h.advance_deadline(
+        &mut rt,
+        CronConfig {
+            continued_faults_penalty: BigInt::from(0), // fee is instead added to debt
+            ..Default::default()
+        },
+    );
+
+    st = h.get_state(&rt);
+    assert_eq!(ff, st.fee_debt);
+
+    // Recovery fails when it can't pay back fee debt
+    expect_abort_contains_message(
+        ExitCode::USR_INSUFFICIENT_FUNDS,
+        "unlocked balance can not repay fee debt",
+        h.declare_recoveries(
+            &mut rt,
+            dl_idx,
+            p_idx,
+            BitField::try_from_bits([one_sector[0].sector_number]).unwrap(),
+            BigInt::from(0),
+        ),
+    );
+
+    // Recovery pays back fee debt and succeeds
+    let funds = &ff + st.initial_pledge + st.locked_funds + st.pre_commit_deposits;
+    rt.set_balance(funds);
+    h.declare_recoveries(
+        &mut rt,
+        dl_idx,
+        p_idx,
+        BitField::try_from_bits([one_sector[0].sector_number]).unwrap(),
+        ff,
+    )
+    .unwrap();
+
+    let dl = h.get_deadline(&rt, dl_idx);
+    let p = dl.load_partition(&rt.store, p_idx).unwrap();
+    assert_eq!(p.faults, p.recoveries);
+    st = h.get_state(&rt);
+    assert_eq!(BigInt::from(0), st.fee_debt);
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn recovery_fails_during_active_consensus_fault() {
+    let mut h = ActorHarness::new(PERIOD_OFFSET);
```

Comment:
This init snippet is repeated in all of the tests, would be nice to extract it to a `setup` function like in some other test suites.